src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py

```
@@ -15,7 +37,374 @@ class CnfNfdGenerator(NFDGenerator):
+            elif hasattr(v, "items") and v.items():
+                # handling schema having properties which doesn't map directly to the values file nesting
+                if "properties" in schema_nested_dict.keys():
+                    schema_nested_dict = schema_nested_dict["properties"][k]
+                else:
+                    schema_nested_dict = schema_nested_dict[k]
+                # recursively call function with values (i.e the nested dictionary)
+                self.find_deploy_params(v, schema_nested_dict, final_schema)
+                # reset the schema dict to its original value (once finished with that level of recursion)
+                schema_nested_dict = original_schema_nested_dict
+
+        return final_schema
+
+    def get_chart_name_and_version(
+        self, helm_package: HelmPackageConfig
+    ) -> Tuple[str, str]:
+        chart = os.path.join(
+            self.tmp_folder_name, helm_package.name, "Chart.yaml"
+        )
+
+        with open(chart, "r", encoding="utf-8") as f:
+            data = yaml.load(f, Loader=yaml.FullLoader)
+            chart_name = data["name"]
+            chart_version = data["version"]
+
+        return (chart_name, chart_version)
+
+    def generate_parmeter_mappings(
+        self, helm_package: HelmPackageConfig
+    ) -> str:
+        # Basically copy the values.mappings.yaml file to the right place.
+        values = os.path.join(
+            self.tmp_folder_name, helm_package.name, "values.mappings.yaml"
+        )
+
+        mappings_folder_path = os.path.join(
+            self.tmp_folder_name, "configMappings"
+        )
+        mappings_filename = f"{helm_package.name}-mappings.json"
+
+        if not os.path.exists(mappings_folder_path):
+            os.mkdir(mappings_folder_path)
+
+        mapping_file_path = os.path.join(
+            mappings_folder_path, mappings_filename
+        )
+
+        with open(values, "r", encoding="utf-8") as f:
```

need to throw a good error to the user if it doesn't exist

---

This is checked elsewhere and will never reach this code if the file doesnt exist